cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -19,11 +19,13 @@
 import networkx as nx
 
 from cirq import protocols
+from cirq.value import value_equality
 
 if TYPE_CHECKING:
     import cirq
 
 
+@value_equality
```

Comment:
this will enable hashing by default which will likely fail because your _value_equality_values_ include dictionaries which are unhashable. Only immutable objects should be hashable. Consider adding `unhashable=True` to the decorator.